src/main/resources/org/jpeek/metrics/TLCOM.xsl

```diff
@@ -0,0 +1,147 @@
+          when a m1 calls m2 and m2 happens to use attribute a2 of the class. This
+          then means that m1 transitively uses a2 via m2.
+          The chain of calls can have more than one hop: if m1 calls m2 which calls
+          m3 which calls m4, and m4 uses attribute a4, then m1 is said to transitively
+          use a4 as well.
+          The final formula is the same as the original LCOM:
+          TLCOM = |P| - |Q|, if |P| &gt; |Q|, otherwise it's 0 (zero). 'P' is
+          the set of method-pairs that are not connected according to the
+          definition above, and 'Q' is the set of method-pairs that are connected.
+        </xsl:text>
+      </description>
+      <xsl:apply-templates select="node()"/>
+    </metric>
+  </xsl:template>
+  <xsl:template match="class">
+    <!--
+    @todo #65:30min TLCOM: we are including access to ALL fields here. After #156 is fixed,
+     come back and refactor xpath for $Ms_that_use_attrs, $left_attrs, and $right_attrs
+     in order to avoid including access to fields belonging to other classes.
+    -->
+    <xsl:variable name="class_fqn" select="replace(string-join(../@id | @id, '.'), '^\.', '')"/>
+    <xsl:variable name="A" select="attributes/attribute"/>
+    <xsl:variable name="M" select="methods/method"/>
+    <xsl:variable name="Ms_that_use_attrs">
+      <xsl:for-each select="$M">
+        <xsl:if test="./ops/op[@code != 'call'][. = $A]">
+          <method>
+            <xsl:value-of select="concat($class_fqn, '.', @name)"/>
+          </method>
+        </xsl:if>
+      </xsl:for-each>
+    </xsl:variable>
+    <xsl:variable name="pairs">
+      <xsl:for-each select="$M">
+        <xsl:variable name="left" select="."/>
+        <xsl:variable name="left_name" select="concat($class_fqn, '.', $left/@name)"/>
+        <xsl:variable name="left_attrs" select="$left/ops/op[@code != 'call' and exists(. = $A)]"/>
+        <xsl:variable name="left_calls" select="$left/ops/op[@code = 'call' and matches(., concat('^', $class_fqn, '\.[^\.]+$'))]"/>
+        <xsl:for-each select="$left/following-sibling::method">
+          <xsl:variable name="right" select="."/>
+          <xsl:variable name="right_name" select="concat($class_fqn, '.', $right/@name)"/>
+          <xsl:variable name="right_attrs" select="$right/ops/op[@code != 'call' and exists(. = $A)]"/>
+          <xsl:variable name="right_calls" select="$right/ops/op[@code = 'call' and matches(., concat('^', $class_fqn, '\.[^\.]+$'))]"/>
+          <!--
+          @todo #65:30min TLCOM: need to establish transitive connections between methods with arbitrarily long
+           chains of method calls between them. Right now, this code can establish transitive connections to a
+           max depth of 3 levels, ie up to something like this: m1 calls m2 calls m3 which uses a3 therefore m1 is
+           connected to m3. It will miss connections in longer chains. After fixing this, maybe we can add it to
```

Comment:
@llorllale I think you wanted to say "m1 is connected to a3", not to m3. Either that, or the "which uses a3" is useless, it causes confusion :D

Reply:
@amihaiemil no, we're creating relations between methods so "m1 is connected to m3" is accurate. This metric says they're connected because m1 is transitively using a3 via indirect method calls all the way to m3. Make sense?

Reply:
@llorllale Well, yes, if ``m1 -> m2 -> m3`` and ``m3 -> a3``, then, according to this metric, ``m1`` is connected to ``a3``; makes sense. But in the puzzle, why did you say "therefore m1 is connected to m3" -- shouldn't it be "m1 is connected to a3"?

Reply:
@amihaiemil This metric says that in this scenario *m1 transitively uses a3* and since m3 also uses a3 then m1 is connected to m3.

Reply:
@llorllale I see the twist now, thanks